java-manta-client/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1431,24 +1431,21 @@ public void putDirectory(final String rawPath, final boolean recursive,
             return;
         }
 
-        final String[] parts = rawPath.split(SEPARATOR);
-        final Iterator<Path> itr = Paths.get("", parts).iterator();
-        final StringBuilder sb = new StringBuilder(SEPARATOR);
-
-        for (int i = 0; itr.hasNext(); i++) {
-            final String part = itr.next().toString();
-            sb.append(part);
+        final Integer skipDepth = config.getSkipDirectoryDepth();
+        final RecursiveDirectoryCreationStrategy directoryCreationStrategy;
+        if (skipDepth != null && 0 < skipDepth) {
+            directoryCreationStrategy = new DepthSkippingDirectoryCreationStrategy(this, skipDepth);
```

Comment:
Is there a way we author the strategy such that we don't need to create a new object per put operation?

Reply:
If the extra object per recursive-`PUT` is a serious concern we could convert the strategies to static methods that use their return value to convey the number of operations taken. This would leave us with just the `RecursiveDirectoryCreationStrategy` class having no constructor and two methods:

`int complete(MantaClient, String, MantaHttpHeaders)`
`int skippingDepth(MantaClient, String, MantaHttpHeaders, int)`